core/templating/datasource_query.go

```diff
@@ -0,0 +1,331 @@
+func parseColumns(columnsPart string, headers []string) []string {
+	columnsPart = strings.TrimSpace(columnsPart)
+	if columnsPart == "*" {
+		return headers
+	}
+	columns := strings.Split(columnsPart, ",")
+	for i, column := range columns {
+		columns[i] = strings.TrimSpace(column)
+	}
+	return columns
+}
+
+// TrimQuotes trims matching single or double quotes from the outer edges of a string.
+func trimQuotes(s string) string {
+	// Check if the string length is less than 2, in which case there's nothing to trim.
+	if len(s) < 2 {
+		return s
+	}
+
+	// Get the first and last characters of the string.
+	firstChar := s[0]
+	lastChar := s[len(s)-1]
+
+	// Check if both the first and last characters are matching quotes.
+	if (firstChar == '\'' || firstChar == '"') && firstChar == lastChar {
+		return s[1 : len(s)-1]
+	}
+
+	// If no trimming is necessary, return the original string.
+	return s
+}
+
+// parseSetClauses parses the SET part of an UPDATE query
+func parseSetClauses(setPart string) map[string]string {
+	setClauses := make(map[string]string)
+	parts := strings.Split(setPart, ",")
+	for _, part := range parts {
+		keyValue := strings.Split(strings.TrimSpace(part), "=")
+		if len(keyValue) == 2 {
+			setClauses[strings.TrimSpace(keyValue[0])] = trimQuotes(strings.TrimSpace(keyValue[1]))
+		}
+	}
+	return setClauses
+}
+
+// parseConditions parses the WHERE part of the query into a slice of Conditions and returns an error if any issues are found.
+func parseConditions(wherePart string) ([]Condition, error) {
+	conditions := []Condition{}
+
+	conditionRegex := regexp.MustCompile(`(\w+)\s*(==|!=|<=|>=|<|>)\s*'([^']*)'`)
+	conditionMatches := conditionRegex.FindAllStringSubmatch(wherePart, -1)
+
+	if len(conditionMatches) == 0 {
+		return conditions, nil
+	}
+
+	for _, match := range conditionMatches {
+		if len(match) != 4 {
+			return nil, errors.New("invalid condition format")
+		}
+		conditions = append(conditions, Condition{
+			Column:   match[1],
+			Operator: match[2],
+			Value:    match[3],
+		})
+	}
+
+	return conditions, nil
+}
+
+// ExecuteSelectQuery executes a SELECT query and returns the results as a slice of RowMaps
+func executeSelectQuery(data [][]string, query SQLStatement) []RowMap {
+	headers := data[0] // First row as header
+	results := []RowMap{}
+	for _, row := range data[1:] {
+		rowMap := mapRow(headers, row)
+		if matchesConditions(rowMap, query.Conditions) {
+			results = append(results, projectRow(rowMap, query.Columns))
+		}
+	}
+	return results
+}
+
+// ExecuteUpdateQuery executes an UPDATE query and modifies the data in-place
+func executeUpdateQuery(data *[][]string, query SQLStatement) error {
+	if len(*data) < 2 {
+		return errors.New("no data available to update")
+	}
+
+	headers := (*data)[0]
+	conditions := query.Conditions
+	setClauses := query.SetClauses
+	for i, row := range (*data)[1:] {
+		rowMap := mapRow(headers, row)
```

Comment:
I started to think that it might be better to store the data as `[]map[string]string` rather than `[][]string`  if we are doing a lot of querying.

Reply:
Agreed - might be worth doing

Reply:
let's leave it for another PR.